packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -54,6 +57,73 @@ List<String> debugDescribeTransform(Matrix4 transform) {
   return matrix;
 }
 
+bool _extensionsInitialized = false;
+
+void initServiceExtensions() {
+  if (_extensionsInitialized)
+    return;
+
+  registerExtension('flutter', _flutter);
+  // TODO: Expose debugDumpApp().
+  // TODO: Expose StatisticsOverlay.
+  registerExtension('flutter.debugPaint', _debugPaint);
+  registerExtension('flutter.timeDilation', _timeDilation);
+
+  _extensionsInitialized = true;
+
+  // Emit an info level log message; this tells the debugger that the Flutter
+  // service extensions are registered.
+  log('Flutter initialized', name: 'flutter', level: 800);
+}
+
+/// Just respond to the request. Clients can use the existence of this call to
+/// know that the debug client is a Flutter app.
+Future<ServiceExtensionResponse> _flutter(String method, Map parameters) {
+  String result = JSON.encode({ 'type': '_extensionType', 'method': method });
+  return new Future.value(new ServiceExtensionResponse.result(result));
+}
+
+/// Toggle the [debugPaintSizeEnabled] setting.
+Future<ServiceExtensionResponse> _debugPaint(String method, Map parameters) {
+  if (parameters.containsKey('enabled')) {
+    // TODO: This is a work around for a VM bug: sdk/25208 - all params are
+    // coerced to strings.
+    // bool enabled = parameters['enabled'] == true;
+    debugPaintSizeEnabled = parameters['enabled'].toString() == 'true';
+    // TODO: How to cause a redraw?
```

Comment:
I guess you could walk the entire tree with visitChildren() calling markNeedsPaint() on everyone...